src/db/oracledb/addresses-dao.js

```diff
@@ -0,0 +1,38 @@
+import _ from 'lodash';
+
+import { serializeAddresses } from 'serializers/addresses-serializer';
+import { parseQuery } from 'utils/parse-query';
+import { getConnection } from './connection';
+import { contrib } from './contrib/contrib';
+
+/**
+ * Queries data source for raw address data and passes it to the serializer before returning
+ *
+ * @param {string} osuId OSU ID of person to select addresses from
+ * @param {object} query Query parameters passed in with request
+ * @returns {Promise<object>} Serialized address resource
+ */
+const getAddressesById = async (osuId, query) => {
+  const connection = await getConnection();
+  try {
+    const parsedQuery = parseQuery(query);
+    parsedQuery.osuId = osuId;
+    parsedQuery.addressType = parsedQuery.addressType ? parsedQuery.addressType : null;
+    parsedQuery.city = parsedQuery.city ? parsedQuery.city : null;
+    parsedQuery.county = parsedQuery.county ? parsedQuery.county : null;
+    parsedQuery.stateCode = parsedQuery.stateCode ? parsedQuery.stateCode : null;
+    parsedQuery.nationCode = parsedQuery.nationCode ? parsedQuery.nationCode : null;
+
+    const { rows } = await connection.execute(contrib.getAdresses(), parsedQuery);
+
+    if (_.isEmpty(rows)) {
+      return undefined;
+    }
+
+    return serializeAddresses(rows, query, osuId);
```

Comment:
Instead of having the DAO return the result of the serializer, I think it should return the raw data. Then, the path handler can pass the result to the serializer. This will help us avoid things like when we want to pass arguments to the serializer that the DAO doesn't need. It also modularizes things a little more and makes the testing a little simpler too.